src/streamlink/plugins/live_russia_tv.py

```diff
@@ -11,71 +11,102 @@
 
 
 class LiveRussia(Plugin):
-    url_re = re.compile(r"https?://(?:www\.|live\.)?russia.tv")
-    _data_re = re.compile(r"""window\.pl\.data\.([\w_]+)\s*=\s*['"]?(.*?)['"]?;""")
+    url_re = re.compile(
+        r"https?://(?:live\.)?russia\.tv/(channel/([0-9]+))?")
+    _data_re = re.compile(
+        r"""window\.pl\.data\.([\w_]+)\s*=\s*['"]?(.*?)['"]?;""")
+    channel = ''
 
     @classmethod
     def can_handle_url(cls, url):
         return cls.url_re.match(url) is not None
 
+
     def _get_iframe_url(self, url):
-        res = self.session.http.get(url)
-        for iframe in itertags(res.text, 'iframe'):
-            src = iframe.attributes.get("src")
-            if src:
-                return src
+        live = self.url_re.match(self.url)
+        channel = live.group(2)
+        log.debug("Channel: {0}".format(channel))
+        if channel: 
+            self.channel = channel
+            template = "https://live.russia.tv/api/now/channel/{0}"
+            return template.format(channel)
+        else:
+            res = self.session.http.get(url)
+            for iframe in itertags(res.text, 'iframe'):
+                src = iframe.attributes.get("src")
+                if src:
+                    return src
+
 
     def _get_stream_info_url(self, url):
-        data = {}
         res = self.session.http.get(url)
-        for m in self._data_re.finditer(res.text):
-            data[m.group(1)] = m.group(2)
-
-        log.debug("Got pl_data={0}".format(data))
+        data = {}
+        if self.channel:
+            args = dict(parse_qsl(urlparse(url).query))
+            if not args:
+                data = self.session.http.json(res)
+                args["id"] = data["live_id"]
+                args["sid"] = "rutv"
+            template = '''
+                https://player.vgtrk.com/iframe/datalive/id/{id}/sid/{sid}'''
+            stream_info_url = template.format(**args)
+        else:
+            for m in self._data_re.finditer(res.text):
+                data[m.group(1)] = m.group(2)
 
-        if data:
             if data["isVod"] == '0':
-                return "https:{domain}/iframe/datalive/id/{id}/sid/{sid}".format(**data)
+                tamplate = "https:{domain}/iframe/datalive/id/{id}/sid/{sid}"
+                stream_info_url = tamplate.format(**data)
             else:
-                return "https:{domain}/iframe/datavideo/id/{id}/sid/{sid}".format(**data)
-        else:
-            args = dict(parse_qsl(urlparse(url).query))
-            return "https://player.vgtrk.com/iframe/datalive/id/{id}/sid/{sid}".format(**args)
+                template = "https:{domain}/iframe/datavideo/id/{id}/sid/{sid}"
+                stream_info_url = template.format(**data)
+
+        log.debug("Got pl_data:")
+        for key in data:
+            log.debug(u'    {0}: {1}'.format(key, data[key]))
```

Comment:
seems like a bit to much spam

only one message `log.trace('{0!r}'.format(data))` might be better